source/js/searcher.js

```diff
@@ -0,0 +1,771 @@
+     * @returns {String} The constucted HTML element containing the result.
+     */
+    buildResultItem: (args) => {
+        const link = args[0];
+        let linkName = F.getPossibleSiteName(link, "/html/");
+        const previews = args[1];
+        let lItem = "<li class=\"searcher-result-item\">";
+        let lHref = `<a href="${link}" class="searcher-result-link">${linkName}`;
+        let lPrev = "";
+        for (i in previews) {
+            lPrev += `<span class="searcher-result-preview">${previews[i]}</span>`;
+        }
+        lItem += lHref + lPrev + "</a></li>";
+        return lItem;
+    },
+    /**
+     * Method used for constructing the whole result list using the result from the search method.
+     * @param {[[String, [String]]]} objs The search results.
+     * @returns {String} The constructed list.
+     */
+    buildResultListFull: (objs) => {
+        let rList = [];
+        for (i in objs) {
+            rList.push(F.buildResultItem(objs[i]));
+        }
+        if (!rList) return "";
+        return F.buildResultList(rList);
+    },
+    /**
+     * Method used for displaying the error message when the website was opened as a file.
+     */
+    urlSchemeErrorMsg: () => console.error("Searcher API will be disabled, because the url scheme must be http or https for CORS request."),
+    /**
+     * Method used for handling the current url scheme.
+     * @param {Boolean} displayMessage If an error occurres display the error message in the console.
+     * @returns True if the current url scheme is HTTP/S, otherwise, false.
+     */
+    urlSchemeCheck: (displayMessage = false) => {
+        if (C.CURRENT_URL_SCHEME_CORRECT) return true;
+        if (displayMessage) F.urlSchemeErrorMsg();
+        return false;
+    },
+    /**
+     * Alias function for the method `console.log(message, ...optionalParams)`.
+     * @param {*} message The message to display.
+     * @param  {...any} args Optional arguments to be displayed with the message.
+     */
+    print: (message, ...args) => args.length > 0 ? console.log(message, args) : console.log(message),
+    /**
+     * Method used for highlighting the search text found in the text provided.
+     * @param {String} text The text provided.
+     * @param {String} search The search text being searched for in the text.
+     * @param {String} backgroundcolor The `background-color` to be applied to the highlighted search value.
+     * @param {String} color The `color` to be used for the highlighted search value.
+     * @returns The highlighted `innerHTML` text.
+     */
+    highlightSearchText: (text, search, backgroundcolor = "navy", color = "white") => {
+        if (F.isEmpty(text) || F.isEmpty(search)) return "";
+        let newText = "";
+        for (let i = 0; i < text.length - search.length; i++) {
+            if (text[i] === "<") {
+                const fCB = text.indexOf(">", i);
```

Comment:
`fCB`?

Reply:
First Closing Bracket.

Added comment and description.